apps/speed.c

```diff
@@ -3571,9 +3576,8 @@ int speed_main(int argc, char **argv)
             }
 
             if (kem_type == KEM_RSA) {
-                bits = atoi(kem_name + 3);
                 params[0] = OSSL_PARAM_construct_size_t(OSSL_PKEY_PARAM_RSA_BITS,
-                                                        &bits);
+                                                        (size_t *)&bits);
```

Comment:
Presumably this (and line 3761) are going to fail badly if `sizeof(unsigned int`)` != `sizeof(size_t)`?

Reply:
A "bad fail" would only happen if sizeof(unsigned int) > sizeof(size_t), no? And that cannot be the case, right?

Reply:
Well, it would depend on endianness, too. Generally, casting a pointer to an integer of one size to a pointer to an integer of another size is a bad thing.

Of course, it might be possible to use `OSSL_PARAM_construct_uint()` here, but I don't know the code well enough

Reply:
Oh dear -- you are absolutely right: `OSSL_PKEY_PARAM_RSA_BITS` indeed is a [`uint` param](https://www.openssl.org/docs/man3.0/man7/RSA.html). Completely silly to have used the `...size_t` constructor :-( Thanks! Fixed in https://github.com/openssl/openssl/pull/20898/commits/4c3102d1d0320be3f3e62fb389a9dab55603f082.

Reply:
The rule is to use the constructor based on the type of the variable you're setting - i.e., no casting. The get_param() call inside the provider will convert appropriately.